### PR TITLE
Update docs and requirements-dev.txt to reflect actual build instructions

### PR DESCRIPTION
**Pull request by vsalvino**
Been a while since I touched this project. Trying to follow the development docs did not work on the first try. Updated the docs and requirements-dev.txt to reflect recent changes.

### CODE REVIEW

**Review by laymonage — COMMENTED** [review]
Related: #225

Rather than referring to `requirements.txt`, we should instead move its contents to the project's dependencies.

**Comment by vsalvino** [comment]
The requirements.txt are used to build the docs (i.e. the docs for the theme). So they are not actually a dependency of the theme, but since the docs are used to test the theme, it is relevant.

**Comment by laymonage** [comment]
> The requirements.txt are used to build the docs (i.e. the docs for the theme). So they are not actually a dependency of the theme, but since the docs are used to test the theme, it is relevant.

Ah, good point. But the `sphinx_copybutton` should be defined in the project itself, though.

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Install package into itself.
 -e .
 
+# Install main requirements.
+-r requirements.txt
```
Comment (laymonage) [inline requirements-dev.txt:5]: After rebasing, the following line will need to be removed:

https://github.com/wagtail/sphinx-wagtail-theme/blob/0dad2f03c5558fd44aacfb8629d4066008af4335/Makefile#L59

**Review by laymonage — COMMENTED** [review]
Just need one tweak to the `Makefile` after #286 was merged, but otherwise this looks good to me! I'll rebase this and merge.

**Review by laymonage — APPROVED** [review]

**Pull request merged**